File: GeneratedEntityReaderClasses/Components/LoadEntitiesComponent.cs
namespace GeneratedNoitaClasses;
public class LoadEntitiesComponent : NoitaComponentBase,  iNoitaType<LoadEntitiesComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaString entity_file { get; set; } = new();
	public ValueRangeInt count { get; set; } = new();
	public NoitaBool kill_entity { get; set; } = new();
	public NoitaInt timeout_frames { get; set; } = new();
	public NoitaInt mTimerTriggerFrame { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		entity_file.Read(s);
		count.Read(s);
		kill_entity.Read(s);
		timeout_frames.Read(s);
		mTimerTriggerFrame.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		entity_file.Write(s);
		count.Write(s);
		kill_entity.Write(s);
		timeout_frames.Write(s);
		mTimerTriggerFrame.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/LoadEntitiesComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from GeneratedClasses.ValueRangeInt import ValueRangeInt
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class LoadEntitiesComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    entity_file:NoitaString = field(default_factory=NoitaString)
    count:ValueRangeInt = field(default_factory=ValueRangeInt)
    kill_entity:NoitaBool = field(default_factory=NoitaBool)
    timeout_frames:NoitaInt = field(default_factory=NoitaInt)
    mTimerTriggerFrame:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.entity_file.Read(s)
        self.count.Read(s)
        self.kill_entity.Read(s)
        self.timeout_frames.Read(s)
        self.mTimerTriggerFrame.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.entity_file.Write(s)
        self.count.Write(s)
        self.kill_entity.Write(s)
        self.timeout_frames.Write(s)
        self.mTimerTriggerFrame.Write(s)

